### PR TITLE
API and List

### PR DESCRIPTION
**Pull request by Marcel-FR**
This is the list of the bestiary.
It shows the names of all monsters.

Fixed the performance issues! 14000+ms fetching time to -> 120-220ms

[Preview](https://capstone-project-git-api-and-list-marcel-fr.vercel.app)

[List US](https://github.com/Marcel-FR/capstone-project/issues/3), [API US](https://github.com/Marcel-FR/capstone-project/issues/5)

### CODE REVIEW

File: resources/lib/dnd-api.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  const monsters = await Promise.all(
+    data.results.map(async (monster) => {
+      const response = await fetch(`https://www.dnd5eapi.co${monster.url}`);
+      const data = await response.json();
+      return { 
+        name: monster.name, 
+        challenge_rating: data.challenge_rating, 
+        type: data.type 
+      };
+    })
```
Comment (Kheirah) [inline resources/lib/dnd-api.js]: That's the reason why the initial fetch takes several seconds to complete. After you fetch all monsters from `"https://www.dnd5eapi.co/api/monsters"` you then loop through ALL these monsters and initiate another fetch to retrieve more info about each monster. In effect, you are fetching 335 times! Once for the initial data and once for each of the 334 monsters.

To fix that, you can simply avoid fetching the additional info from each monster. You probably don't need to display their details on the bestiary page and can instead defer the fetch for detailed info into the detail page of each monster. That way, you're never fetching 335x.

**Review by Kheirah — CHANGES_REQUESTED** [review]
Very good work!

You only need to think about fetching data in order to reduce loading times. See if you can change some code around with the help of my comment below.

File: resources/lib/dnd-api.js (hunk 1 above)
Comment (Marcel-FR) [reply resources/lib/dnd-api.js]: I fixed that issue now.

**Review by MelissaDuve — APPROVED** [review]
nice!

**Pull request merged**